app/models/compute_resources/foreman/model/ec2.rb

```diff
@@ -51,6 +51,7 @@ def create_vm(args = { })
       args[:groups].reject!(&:empty?) if args.has_key?(:groups)
       args[:security_group_ids].reject!(&:empty?) if args.has_key?(:security_group_ids)
       args[:associate_public_ip] = subnet_implies_is_vpc?(args) && args[:managed_ip] == 'public'
+      args[:private_ip_address] = args[:interfaces_attributes][:"0"][:ip]
```

Comment:
How does this handle the case when the VM has no ip set by Foreman, i.e. when EC2 provides the ip? It might not be a problem, just asking.